Menus/MainMenuGUI.cpp

```diff
@@ -1,3263 +1,539 @@
-        else if (m_ConfigureStep == 2)
-        {
-            m_pConfigLabel[CONFIGINPUT]->SetText("MOVE LEFT");
-            m_pConfigLabel[CONFIGRECOMMENDATION]->SetText("[A]");
-            if (g_UInputMan.CaptureKeyMapping(m_ConfiguringPlayer, INPUT_L_LEFT))
-            {
-                g_UInputMan.CaptureKeyMapping(m_ConfiguringPlayer, INPUT_R_LEFT);
-                m_ConfigureStep++;
-                m_ScreenChange = true;
-            }
-        }
-        // Move right
-        else if (m_ConfigureStep == 3)
-        {
-            m_pConfigLabel[CONFIGINPUT]->SetText("MOVE RIGHT");
-            m_pConfigLabel[CONFIGRECOMMENDATION]->SetText("[D]");
-            if (g_UInputMan.CaptureKeyMapping(m_ConfiguringPlayer, INPUT_L_RIGHT))
-            {
-                g_UInputMan.CaptureKeyMapping(m_ConfiguringPlayer, INPUT_R_RIGHT);
-                m_ConfigureStep++;
-                m_ScreenChange = true;
-            }
-        }
-        // Move right
-        else if (m_ConfigureStep == 4)
-        {
-            m_pConfigLabel[CONFIGINPUT]->SetText("RELOAD");
-            m_pConfigLabel[CONFIGRECOMMENDATION]->SetText("[R]");
-            if (g_UInputMan.CaptureKeyMapping(m_ConfiguringPlayer, INPUT_WEAPON_RELOAD))
-            {
-                m_ConfigureStep++;
-                m_ScreenChange = true;
-            }
-        }
-		else if (m_ConfigureStep == 5)
-		{
-			m_pConfigLabel[CONFIGINPUT]->SetText("PICK UP");
-			m_pConfigLabel[CONFIGRECOMMENDATION]->SetText("[F]");
-			if (g_UInputMan.CaptureKeyMapping(m_ConfiguringPlayer, INPUT_WEAPON_PICKUP))
-			{
-				m_ConfigureStep++;
-				m_ScreenChange = true;
-			}
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool MainMenuGUI::RollCredits() {
+		int scrollDuration = m_CreditsTextLabel->GetHeight() * 50;
+		float scrollDist = static_cast<float>(m_CreditsScrollPanel->GetHeight() + m_CreditsTextLabel->GetHeight());
+		float scrollProgress = static_cast<float>(m_CreditsScrollTimer.GetElapsedRealTimeMS()) / static_cast<float>(scrollDuration);
+		m_CreditsTextLabel->SetPositionRel(0, m_CreditsScrollPanel->GetHeight() - static_cast<int>(scrollDist * scrollProgress));
+
+		if (m_CreditsScrollTimer.IsPastRealMS(scrollDuration + 1000)) {
+			return true;
 		}
-		else if (m_ConfigureStep == 6)
-		{
-			m_pConfigLabel[CONFIGINPUT]->SetText("DROP");
-			m_pConfigLabel[CONFIGRECOMMENDATION]->SetText("[G]");
-			if (g_UInputMan.CaptureKeyMapping(m_ConfiguringPlayer, INPUT_WEAPON_DROP))
-			{
-				m_ConfigureStep++;
-				m_ScreenChange = true;
-			}
+		return false;
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	MainMenuGUI::MainMenuUpdateResult MainMenuGUI::Update() {
+		m_UpdateResult = MainMenuUpdateResult::NoEvent;
+
+		if (g_ConsoleMan.IsEnabled() && !g_ConsoleMan.IsReadOnly()) {
+			return m_UpdateResult;
```

Comment:
I just think it's pointless to handle the menus while the console is active so i'm early returning if it's not in read-only mode.

Reply:
It's pointless cause consoleman will be blocking all input right? Just wanna be sure I'm understanding the reasoning here.

Reply:
It won't be blocking anything.
It will be conflicting when trying to interact with elements that require text input, but other than that nothing that I can think of.

It's just generally pointless.
Give me one example of needing to be able to interact with menus while actively inputting to the console.

Reply:
As discussed in pms, I'm fine with this behaviour. Feel free to resolve.